### DISCUSSION TITLE
how to get the MetaData by Driver Manager

### DISCUSSION THREAD

**Discussion by EurekaJiang** [dbeaver/dbeaver#18219] [Q&A]
I think it's amazing that when I   "Create new driver" ,setting the driver info and then  add Libraries , connected  ,then I can see the metadata . where is the infomation from ? the libraries? I wonder that  I can connect "my own database" if  I follow the rules.

**Comment by LonwoLonwo** [comment]
I'm not sure, but maybe this interface will benefit you.

https://github.com/dbeaver/dbeaver/blob/devel/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/exec/jdbc/JDBCDatabaseMetaData.java

Most of the metadata requests pass through it.

**Reply by EurekaJiang** [reply]
yeah, I just noticed that there is no trigger defined in this interface, so when I used the Driver Manager to connect the database which has triggers , it won't show the trigger list. your advice is very helpful, thank you!

and there is an idea for the function "create new driver", maybe the user can upload the database icon so the database can be seen in the dblist at one glance like mysql, oracle....

**Reply by LonwoLonwo** [reply]
That's because DatabaseMetaData does not have getTriggers method. 
Because not all databases have triggers.
In DBeaver we are reading triggers directly from database without driver help.

**Comment by EurekaJiang** [comment]
I added  break points , the metadata request pass through the interface , and it gets the metadata from  jdbcdriver,  in the database libraries,it offered the function to select the database metadata(there are many SELECT statement in the DatabaseMetaData class) . So, if I write my own database driver classes  that support the jdbc ,or odbc...., it will be connected.
